Framework/PythonInterface/mantid/api/src/Exports/SpectrumDefinition.cpp

```diff
@@ -0,0 +1,36 @@
+#include "MantidTypes/SpectrumDefinition.h"
+
+#include <boost/python/class.hpp>
+#include <boost/python/tuple.hpp>
+
+using Mantid::SpectrumDefinition;
+using namespace boost::python;
+
+// Helper function to convert a std::Pair to a Python tuple
+boost::python::tuple convertPairToTuple(const SpectrumDefinition &self,
+                                        const size_t index) {
+  const std::pair<size_t, size_t> pair = self.operator[](index);
+  boost::python::tuple ret_val = make_tuple(pair.first, pair.second);
+  return ret_val;
+}
+
+void export_SpectrumDefinition() {
+  class_<SpectrumDefinition>("SpectrumDefinition", no_init)
+
+      .def("size", &SpectrumDefinition::size, arg("self"),
+           "Returns the size of the SpectrumDefinition i.e. the number of "
+           "detectors for the spectrum.")
+
+      .def("add", &SpectrumDefinition::add,
+           (arg("self"), arg("detectorIndex"), arg("timeIndex")),
+           "Adds a pair of detector index and time index to the spectrum "
+           "definition.")
+
+      .def("equals",
+           &SpectrumDefinition::operator==, (arg("self"), arg("other")),
+           "Compare spectrum definitions.")
+
+      .def("get", &convertPairToTuple, (arg("self"), arg("index")),
```

Comment:
I suggest a conservative view with exposing everything.. We should involve @ianbush in this when it comes to step scans.

Reply:
We do have some places where we would like to use `SpectrumInfo` from Python so I am all in favour of exposing the time indexes at this point. For manual testing please take a look at `CreateSampleWorkspace`, you can create scanning workspaces via the `NumScanPoints` property. 

For this to be useful we also need the corresponding access methods in `DetectorInfo`. I wonder if it is worth sketching out how they might look so we can get an idea of what the API will look like as a whole, or is this done somewhere already? I think a tuple for `SpectrumDefinition` makes sense in the Python API. Is it possible to make this a named tuple, so you can get just the `detectorIndex` easily without needing `[0]` everywhere?

Reply:
Hi Ian, that sounds like a good idea! I'll make some plans for what will be exposed for `SpectrumInfo` , `DetectorInfo` and `ComponentInfo` and you can let me know what you think?